[Feature] Restrict the format of variable code in the variable management module #1851

muchunjin opened the pull request:
<!--
Thank you for contributing to StreamPark! Please make sure that your code changes
are covered with tests. And in case of new features or big changes
remember to adjust the documentation.

## Contribution Checklist

  - If this is your first time, please read our contributor guidelines: [Submit Code](https://streampark.apache.org/community/submit_guide/submit_code).

  - Make sure that the pull request corresponds to a [GITHUB issue](https://github.com/apache/streampark/issues).

  - Name the pull request in the form "[Feature] Title of the pull request", where *Feature* can be replaced by `Hotfix`, `Bug`, etc.

  - Fill out the template below to describe the changes contributed by the pull request. That will give reviewers the context they need to do the review.

  - If the PR is unfinished, add `[WIP]` in your PR title, e.g., `[WIP][Feature] Title of the pull request`.

-->

## What changes were proposed in this pull request

Issue Number: close #1851  <!-- REMOVE this line if no issue to close -->

<!--(For example: This pull request proposed to add checkstyle plugin).-->

## Brief change log

<!--*(for example:)*
- *Add maven-checkstyle-plugin to root pom.xml*
-->

## Verifying this change

<!--*(Please pick either of the following options)*-->

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

<!--*(example:)*
- *Added integration tests for end-to-end.*
- *Added *Test to verify the change.*
- *Manually verified the change by testing locally.* -->

## Does this pull request potentially affect one of the following parts
 - Dependencies (does it add or upgrade a dependency): (yes / no)

Commit:
[Feature] Added variable management menu to team admin role #1849

Commit:
[Feature] Restrict the format of variable code in the variable management module #1851

Commit:
Merge remote-tracking branch 'upstream/dev' into add-macksonmu

1996fanrui commented on streampark-console/streampark-console-webapp/src/views/system/variable/Add.vue:
```diff
@@ -127,23 +127,21 @@ export default {
+        }).then((resp) => {
+          if (resp.status !== 'success') {
+            this.validateStatus = 'error'
+            this.help = resp.message
```
The check can be added in front-end. You can refer how to limit the User name length.

wolfboys commented on streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/system/controller/VariableController.java:
```diff
@@ -87,6 +92,11 @@ public RestResponse deleteVariables(@Valid Variable variable) throws Exception {
+        try {
+            this.checkVariableCodeFormat(variableCode);
+        } catch (ApiAlertException e) {
+            return RestResponse.fail(e.getMessage(), ResponseCode.CODE_FAIL_ALERT);
```
we don't need try exception there, just `throw exception`

Commit:
[Feature] Restrict the format of variable code in the variable management module #1851 #1852

muchunjin commented:
@1996fanrui @wolfboys 
I modified three places
- Added the inspection of variable code format in the front end.
- Moved the variable management module to core and adjusted the order of the menus.
- Fixed menu not sorting when admin is logged in.

wolfboys reviewed (approved):
LGTM
